src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,178 @@
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+/**
+ * <p>
+ * Checks if unnecessary null check is used around instanceof expression.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code tokens} - tokens to check
+ * Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_IF">
+ * LITERAL_IF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NullTestAroundInstanceOf&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *
+ *     if (myObj != null &amp;&amp; myObj instanceof Object) { // violation
+ *
+ *     }
+ *
+ *     if (myObj != null) { // violation
+ *         if (myObj instanceof Object) {
+ *
+ *         }
+ *     }
+ *
+ *    if ((myObj != null) &amp;&amp; (myObj instanceof Object)) {}  // violation
+ * }
+ * </pre>
+ *
+ * @since 8.33
+ */
+@StatelessCheck
+public class NullTestAroundInstanceOfCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message null.before.instanceof in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "null.test.around.instanceof";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_IF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST landNode = ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(TokenTypes.LAND);
+        if (landNode == null) {
+            if (checkOnNextLine(ast)) {
+                log(ast, MSG_KEY);
+            }
+        }
+        else {
+            final DetailAST notEqualNode = landNode.findFirstToken(TokenTypes.NOT_EQUAL);
+            final DetailAST instanceOfNode =
+                    landNode.findFirstToken(TokenTypes.LITERAL_INSTANCEOF);
+            if (notEqualNode != null && instanceOfNode != null
+                    && notEqualNode.findFirstToken(TokenTypes.LITERAL_NULL) != null) {
+                final DetailAST dotNode1 = getDotNode(notEqualNode);
+                final DetailAST dotNode2 = getDotNode(instanceOfNode);
+                final String string1;
+                if (dotNode1 == null) {
+                    string1 = getIdentNode(notEqualNode).getText();
```

Comment:
I think here you should recursively get full field name, 
it is possible that field could contain more than one dot, for example  `person.department.name`

Reply:
Probably use `FullIdent`.

Reply:
done, Thanks @rnveach